<launch>

    <!-- parameters for spawn nn model -->
    <arg name="nn_model_path"   default="$(find rosbot_controller)/src/model.onnx" />
    <arg name="cmd_topic"       default="cmd_vel" />
    <arg name="parent_frame"    default="odom"/>
    <arg name="robot_frame"     default="nn_model_link"/>
    <arg name="cmd_freq"        default="30"/>

	<!-- spawn kinematic model -->
    <node pkg="rosbot_controller" type="nn_model_runner.py" name="nn_model_runner" output="screen">
        <param name="model_path"       value="$(arg nn_model_path)" />
        <param name="cmd_topic"        value="$(arg cmd_topic)" />
        <param name="parent_frame"     value="$(arg parent_frame)" />
        <param name="robot_frame"      value="$(arg robot_frame)" />
        <param name="cmd_freq"         value="$(arg cmd_freq)" />
    </node>   

</launch>
